### PR TITLE
APIM-60: add security support on kafka endpoint

### PR DESCRIPTION
**Pull request by guillaumelamirand**
## Issue

https://graviteecommunity.atlassian.net/browse/APIM-60

## Description

Adapt kafka oss endpoint to be able to support security

## Additional context

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/apim-60-kafka-security/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-wgsxcvdpxw.chromatic.com)

### CODE REVIEW

**Review by ytvnr — APPROVED** [review]

File: gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-kafka/src/main/java/io/gravitee/plugin/endpoint/kafka/KafkaEndpointConnector.java [hunk 1]
```diff
@@ -127,6 +130,14 @@ protected Completable publish(final ExecutionContext ctx) {
         }
     }
 
+    /**
+     * This method is used to add custom configuration for producer client
+     * @param config
+     */
+    protected void addCustomProducerConfig(final Map<String, Object> config) {
+        // Nothing to do here
```
Comment (marcambier) [inline gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-kafka/src/main/java/io/gravitee/plugin/endpoint/kafka/KafkaEndpointConnector.java:138]: Can you please explain the point of adding those 2 empty methods ?

File: gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-kafka/src/main/java/io/gravitee/plugin/endpoint/kafka/factory/ClassLoaderHelper.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import org.apache.kafka.common.utils.Utils;
+
+/**
+ * @author Guillaume LAMIRAND (guillaume.lamirand at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class ClassLoaderHelper {
+
+    /**
+     * In order to properly manage class loading while dealing with kafka, we need to erase the current thread classloader by <code>null</code> to force Kafka to use plugin classloader instead of node class loader
```
Comment (marcambier) [inline gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-kafka/src/main/java/io/gravitee/plugin/endpoint/kafka/factory/ClassLoaderHelper.java]: Is it intented to be `<code>null</code>` in the javadoc ?

File: gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-kafka/src/main/java/io/gravitee/plugin/endpoint/kafka/KafkaEndpointConnector.java (hunk 1 above)
Comment (guillaumelamirand) [reply gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-kafka/src/main/java/io/gravitee/plugin/endpoint/kafka/KafkaEndpointConnector.java:138]: It is for adding custom configuration by the advanced endpoint. I have rephrase the javadoc, let me know wdyt.

File: gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-kafka/src/main/java/io/gravitee/plugin/endpoint/kafka/factory/ClassLoaderHelper.java (hunk 2 above)
Comment (guillaumelamirand) [reply gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-kafka/src/main/java/io/gravitee/plugin/endpoint/kafka/factory/ClassLoaderHelper.java]: That was the first implementation, forget to update it, will do. Good catch

File: gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-kafka/src/main/java/io/gravitee/plugin/endpoint/kafka/KafkaEndpointConnector.java (hunk 1 above)
Comment (marcambier) [reply gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-kafka/src/main/java/io/gravitee/plugin/endpoint/kafka/KafkaEndpointConnector.java:138]: Oh, I see, got it !
Yea, a little javadoc rephrase would help

**Review by marcambier — APPROVED** [review]

**Pull request merged**